[Design]  룸 리스트 뷰 디자인 변경 / 시공 상황 작성하기 버튼 색상 변경

creohwan opened the pull request:
## Motivation 🥳 (코드를 추가/변경하게 된 이유)
- 룸 리스트 뷰 디자인 변경
- 시공 상황 작성하기 버튼 색상 변경

## Key Changes 🔥 (주요 구현/변경 사항)

## ToDo 📆 (남은 작업)
- [ ] todo

## ScreenShot 📷 (참고 사진)
![IMG_5979](https://user-images.githubusercontent.com/92836045/206822545-1f86b7d4-edfd-4cb6-bb8d-601b20820c5d.PNG)

## To Reviewers 🙏 (리뷰어에게 전달하고 싶은 말)

## Reference 🔗

## Close Issues 🔒 (닫을 Issue)
Close #163

creohwan committed:
Design:iphone:: #163 Add Design

creohwan committed:
Design:iphone:: #163 Add Design

creohwan committed:
Style:art:: #163 Change Name

creohwan committed:
Style:art:: #163 Change Name

creohwan committed:
Rename:pencil:: #163 Change File Order

creohwan committed:
Rename:pencil:: #163 Change File Order

creohwan committed:
Revert "Rename:pencil:: #163 Change File Order"

This reverts commit 11d08781fbfc749c8608f574787648fef1fe0d23.

creohwan committed:
Revert "Rename:pencil:: #163 Change File Order"

This reverts commit 11d08781fbfc749c8608f574787648fef1fe0d23.

creohwan committed:
Rename:pencil:: #163 Change File Order

creohwan committed:
Rename:pencil:: #163 Change File Order

creohwan committed:
Fix:beetle:: #163 Change Button Color

creohwan committed:
Fix:beetle:: #163 Change Button Color

creohwan committed:
Design:iphone:: #163 Modify Padding

creohwan committed:
Design:iphone:: #163 Modify Padding

creohwan committed:
Design:iphone:: #179 Add Developer View

creohwan committed:
Design:iphone:: #179 Add Developer View

creohwan committed:
Merge branch 'design/179-make-developerview' of https://github.com/DeveloperAcademy-POSTECH/MacC-Team-Samsamhada into design/179-make-developerview

creohwan committed:
Design:iphone:: #179 Change Background Color

JUNY0110 reviewed (changes requested):
Section 간 간격에 차이가 있어야 합니다~  AS관리고객 상단은 24, AS관리고객의 하단은 16입니다.

creohwan committed:
Design:iphone:: #163 Modify Padding

creohwan committed:
Design:iphone:: #163 Modify Padding

CreoHwan commented:
> Section 간 간격에 차이가 있어야 합니다~ AS관리고객 상단은 24, AS관리고객의 하단은 16입니다.

31000ed 감사합니다~!

creohwan committed:
Merge pull request #185 from DeveloperAcademy-POSTECH/design/179-make-developerview

[Design] 개발자 정보 뷰 수정 / 버전 정보 네이게이션 클릭 비활성화

creohwan committed:
Design:iphone:: #163 Modify Layout

creohwan committed:
Design:iphone:: #163 Modify Layout

CreoHwan commented:
> > > Section 간 간격에 차이가 있어야 합니다~ AS관리고객 상단은 24, AS관리고객의 하단은 16입니다.
> > 
> > 
> > [31000ed](https://github.com/DeveloperAcademy-POSTECH/MacC-Team-Samsamhada/commit/31000ed29457fe762ba6344580b2eadc67aa4caf) 감사합니다~!
> 
> CollectionView의 EdgeInset을 80정도 주면 최하단의 방목록이 버튼에 가려지지 않을 것 같습니당

넵 감사합니다~ f78416f

ikanic commented on Samsam/ViewController/RoomListView/RoomListViewController.swift:
```diff
@@ -16,16 +16,29 @@ class RoomListViewController: UIViewController {
         }
     }
+    
+    var doingRooms = [Room]()
+    var doneRooms = [Room]()
```
p4;
변수 이름이 애매하네요..
당장 떠오르는 이름이 없긴하지만, 비직관적이라는 생각이 들긴합니다.

ikanic commented on Samsam/ViewController/RoomListView/RoomListViewController.swift:
```diff
@@ -104,65 +165,66 @@ extension RoomListViewController: UICollectionViewDataSource, UICollectionViewDe
-        cell.startDate.text = convertDate(dateString: rooms[indexPath.row].startDate)
-        cell.endDate.text = convertDate(dateString: rooms[indexPath.row].endDate)
+        
+        if indexPath.section == 0 {
+            tapRoomListButton.rooms = doingRooms[indexPath.item]
+            cell.roomTitle.text = doingRooms[indexPath.row].clientName
+            cell.startDate.text = convertDate(dateString: doingRooms[indexPath.row].startDate)
+            cell.endDate.text = convertDate(dateString: doingRooms[indexPath.row].endDate)
+            cell.chipText.text = "\(cell.startDate.text!) ~ \(String(describing: cell.endDate.text!))"
+        } else {
+            tapRoomListButton.rooms = doneRooms[indexPath.item]
+            cell.roomTitle.text = doneRooms[indexPath.row].clientName
+            cell.startDate.text = convertDate(dateString: doneRooms[indexPath.row].startDate)
+            cell.endDate.text = convertDate(dateString: doneRooms[indexPath.row].endDate)
+            cell.chipText.text = "\(cell.startDate.text!) ~ \(cell.endDate.text!)"
```
p1;
해당 부분은 비슷한 구문이 반복되고 있는 것 같습니다. 해당 부분은
```swift
let newArray = indexPath.section == 0 ? doingRooms : doneRooms

tapRoomListButton.rooms = newArray[indexPath.item]
cell.roomTitle.text = neweArray[indexPath.row].clientName
cell.startDate.text = convertDate(dateString: newArray[indexPath.row].startDate)
cell.endDate.text = convertDate(dateString: newArray[indexPath.row].endDate)
cell.chipText.text = "\(cell.startDate.teext!) ~ \(String(describing: cell.endDate.text!))"
```
와 같이 변경할 수 있겠네요.

또, indexPath.item과 indexPath.row가 혼용되고 있는 것 같은데 이렇게 되어야 하는 이유가 있을까요?

creohwan replied:
> p1; 해당 부분은 비슷한 구문이 반복되고 있는 것 같습니다. 해당 부분은
> 
> ```swift
> let newArray = indexPath.section == 0 ? doingRooms : doneRooms
> 
> tapRoomListButton.rooms = newArray[indexPath.item]
> cell.roomTitle.text = neweArray[indexPath.row].clientName
> cell.startDate.text = convertDate(dateString: newArray[indexPath.row].startDate)
> cell.endDate.text = convertDate(dateString: newArray[indexPath.row].endDate)
> cell.chipText.text = "\(cell.startDate.teext!) ~ \(String(describing: cell.endDate.text!))"
> ```
> 
> 와 같이 변경할 수 있겠네요.
> 
> 또, indexPath.item과 indexPath.row가 혼용되고 있는 것 같은데 이렇게 되어야 하는 이유가 있을까요?

c3038ba 감사합니다~!

ikanic replied:
> > p1; 해당 부분은 비슷한 구문이 반복되고 있는 것 같습니다. 해당 부분은
> > ```swift
> > let newArray = indexPath.section == 0 ? doingRooms : doneRooms
> > 
> > tapRoomListButton.rooms = newArray[indexPath.item]
> > cell.roomTitle.text = neweArray[indexPath.row].clientName
> > cell.startDate.text = convertDate(dateString: newArray[indexPath.row].startDate)
> > cell.endDate.text = convertDate(dateString: newArray[indexPath.row].endDate)
> > cell.chipText.text = "\(cell.startDate.teext!) ~ \(String(describing: cell.endDate.text!))"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 와 같이 변경할 수 있겠네요.
> > 또, indexPath.item과 indexPath.row가 혼용되고 있는 것 같은데 이렇게 되어야 하는 이유가 있을까요?
> 
> [c3038ba](https://github.com/DeveloperAcademy-POSTECH/MacC-Team-Samsamhada/commit/c3038ba53da29f713dd9b6a32179f5c0713d4eed) 감사합니다~!

확인했습니다~

ikanic commented on Samsam/ViewController/RoomListView/RoomListCell.swift:
```diff
@@ -0,0 +1,161 @@
+
+    // MARK: - View
+
+    private let cellStack: UIStackView = {
+        $0.axis = .horizontal
+        return $0
+    }(UIStackView())
+
+    let roomStack: UIStackView = {
+        $0.axis = .horizontal
+        $0.backgroundColor = .white
+        $0.layer.cornerRadius = 16
+        return $0
+    }(UIStackView())
```
p1;
해당 부분은 제가 예전 디자인을 가지고 셀에서 두가지 액션을 주기 위해서 나눠줬던 부분입니다.
현재의 디자인에서는 이렇게 나뉠 이유가 없는 것 같습니다.

creohwan replied:
> p1; 해당 부분은 제가 예전 디자인을 가지고 셀에서 두가지 액션을 주기 위해서 나눠줬던 부분입니다. 현재의 디자인에서는 이렇게 나뉠 이유가 없는 것 같습니다.

24ae2fd 감사합니다~

ikanic replied:
> > p1; 해당 부분은 제가 예전 디자인을 가지고 셀에서 두가지 액션을 주기 위해서 나눠줬던 부분입니다. 현재의 디자인에서는 이렇게 나뉠 이유가 없는 것 같습니다.
> 
> [24ae2fd](https://github.com/DeveloperAcademy-POSTECH/MacC-Team-Samsamhada/commit/24ae2fdfbf6653ea2d301b05fb895d1ce6ae196c) 감사합니다~

확인했습니다. 리뷰 리졸브합니다.

ikanic commented on Samsam/ViewController/RoomListView/RoomListCell.swift:
```diff
@@ -0,0 +1,161 @@
+        return $0
+    }(UIStackView())
+
+    private let leftSpacer: UIView = {
+        $0.setWidth(width: 16)
+        return $0
+    }(UIView())
+
+    private let rightSpacer: UIView = {
+        $0.setWidth(width: 16)
+        return $0
+    }(UIView())
```
p1;
마찬가지로 Spacer들 또한 제가 예전 디자인을 위해 만들어 두었던 부분입니다.
현재 디자인에서는 Spacer 없이 패딩으로 충분히 조절이 가능한 것 같네요.

creohwan replied:
> p1; 마찬가지로 Spacer들 또한 제가 예전 디자인을 위해 만들어 두었던 부분입니다. 현재 디자인에서는 Spacer 없이 패딩으로 충분히 조절이 가능한 것 같네요.

24ae2fd 감사합니다~

ikanic replied:
> > p1; 마찬가지로 Spacer들 또한 제가 예전 디자인을 위해 만들어 두었던 부분입니다. 현재 디자인에서는 Spacer 없이 패딩으로 충분히 조절이 가능한 것 같네요.
> 
> [24ae2fd](https://github.com/DeveloperAcademy-POSTECH/MacC-Team-Samsamhada/commit/24ae2fdfbf6653ea2d301b05fb895d1ce6ae196c) 감사합니다~

확인했습니다. 리뷰 리졸브합니다.

ikanic reviewed (changes requested):
고생하셨습니다.
수정이 필요해보이는 부분 리뷰 남겼습니다. 확인 부탁드려요.

ikanic committed:
!HOTFIX:ambulance:: #189 Not Insert phonenum

- 회원가입시에 전화번호를 입력하지 않은 상태로 강제 종료 후, 다시 앱을 실행하면 방목록으로 이동하고, 그 후에 전화번호를 수정하려해도 앱이 강제 종료되는 버그 해결

ikanic committed:
Chore:wrench:: #189 Modify text space

- 텍스트 띄어쓰기 변경

ikanic committed:
Style:art:: #189 Modify space

- 어레이 내부 스페이스 컨벤션에 맞게 변경

ikanic committed:
Feat:sparkles:: #189 Add twoSelectionAlert

- 두가지 선택지를 가진 경고창을 보여주는 메서드 추가

ikanic committed:
Chore:wrench:: #189 Modify logout func

- 설정 창에서 로그아웃을 눌렀을 때 바로 로그아웃 되지 않고, 경고창을 띄어준 후에 로그아웃 되도록 변경

ikanic committed:
Chore:wrench:: #189 Modify withdrawn

- 설정 창에서 회원탈퇴를 눌렀을 시, 바로 회원탈퇴 되지 않고 경고창을 먼저 보여준 후 탈퇴되도록 변경

ikanic committed:
Merge pull request #191 from DeveloperAcademy-POSTECH/HOTFIX/189-not_insert_phonenum_cannot_modify

[!HOTFIX] 회원가입시 전화번호를 입력하지 않고 어플리케이션을 종료하면 후에 수정이 불가능한 버그 수정

creohwan committed:
Merge branch 'design/163-modify-roomlist' of https://github.com/DeveloperAcademy-POSTECH/MacC-Team-Samsamhada into design/163-modify-roomlist

creohwan committed:
Style:art:: #163 Make More Clear

creohwan committed:
Design:iphone:: #163 Delete

ikanic reviewed (approved):
고생하셨습니다.
변경 사항과 수정 사항이 많아서 힘드셨을텐데, 정말 고생 많으셨어요.

creohwan committed:
Design:iphone:: #184 Change Font
